### ISSUE TITLE
list index out of range: 310

### ISSUE THREAD

**Issue by TamaMcGlinn** [rbong/vim-flog#56]
When trying to open the log using

`:call flogmenu#open_git_log()<CR>`

I got the following error, on two different repositories, after having tried to cherry-pick a stash commit (don't ask me why), and then doing a git stash apply outside of flog. It left the flog buffer uninitialized.

```
Error detected while processing function flogmenu#open_git_log[1]..flog#open[10]..flog#open_graph[6]..flog#initialize_graph_buffer[4]..flog#populate_gr
aph_buffer[10]..flog#parse_log_output:                                                                                                            
line   17:
E684: list index out of range: 310
E15: Invalid expression: [a:output[l:i]]
```

I was 10 commits behind on your master (forked off e27ba02) when I got this, but restarting vim made the problem go away. I wonder if it could be caused by doing git operations outside of flog which make the log shorter, and then trying some operation on the topmost commit in the git log. Perhaps some insight into parse_log_output is necessary.

I am currently unable to reproduce so I recommend closing this. If anyone else has the issue, feel free to reopen (especially if you have repro steps).

**Comment by TamaMcGlinn** [comment]
Hmmm... on my other computer this repo's every time, for every repository, so I have to fix this now.

```
  while l:i < l:output_len && a:output[l:i] !~# g:flog_format_start
    let l:raw += [a:output[l:i]]
    let l:i += 1
  endwhile
  if l:raw != []
    let l:raw += [a:output[l:i]]
    let l:i += 1
  endif
```

It seems that you expect to encounter the `__FSTART__' one before the end. The `index < output_len` evidently never broke the first loop previously, since when that happens, we always will get this error immediately after, since the second access doesn't check that condition before indexing. Just adding `&& l:i < l:output_len` gives an error a little bit later down:

```
Error detected while processing function flogmenu#open_git_log[1]..flog#open[10]..flog#open_graph[6]..flog#initialize_graph_buffer[4]..flog#populate_graph_buffer[12]..flog#set_graph_buffer_commits[3]..flog#get_l
og_display:                                                                                                                                                                                                   
line    3:
E716: Key not present in Dictionary: "display"
E15: Invalid expression: l:c.display
```

So I guess this is just fighting a symptom of the real problem.
```

**Comment by TamaMcGlinn** [comment]
`unlet g:flog_build_log_command_fn` fixed it; so I guess there is a requirement on the git log output that I'm no longer satisfying. :( Do you still use some sort of git-forest script yourself?

**Comment by rbong** [comment]
There are no new requirements for `flog_build_log_command_fn`, I'm not using any custom version of `git-forest`. My custom command still looks like this.

```vim
    function FlogBuildLog() abort
        let l:state = flog#get_state()

        if l:state.no_graph
            return flog#build_log_command()
        endif

        let l:command = 'export GIT_DIR='
        let l:command .= shellescape(flog#get_fugitive_git_dir())
        let l:command .= '; '

        let l:command .= 'git-forest '
        let l:command .= substitute(flog#build_log_args(), ' --graph', '', '')
        let l:command .= ' -- '
        let l:command .= flog#build_log_paths()

        return l:command
    endfunction
    let g:flog_build_log_command_fn = 'FlogBuildLog'
```

Make sure you are using `GIT_DIR`.

If that doesn't fix it, please post steps to recreate. I need at least your settings, I need to know what `flogmenu#open_git_log` is, and if possible I need the repo you can recreate this on.

**Comment by rbong** [comment]
So I see what flogmenu is now, I just need confirmation on what version of the code I should be looking at.

Also the issue with `Key not present in Dictionary: "display"` sounds like an entirely different issue from not checking for length, if you look for `let l:c_display = ...` you'll see that it only doesn't get set in the function if there is no start token in the output. Overall I am very curious to see the command output you're getting even if you otherwise fix this, so please let me know what repository/settings you're using regardless.

**Comment by TamaMcGlinn** [comment]
Yea, I see now that I had tried to simplify things too much, and I guess for some reason (maybe not reloading vimrc) I didn't notice that it has stopped working so I committed this to my vimrc and carried on for a while:

```
 function FlogBuildLog() abort
   return 'forest --style=10'
  endfunction
```

I can repro the bug with that for any repo, no need to use flogmenu.

Thanks for following up, and sorry for taking up time when I could've just reverted my vimrc to fix this. I erroneously thought that this was related to my switching to using [git worktrees](https://www.youtube.com/watch?v=2uEqYw-N8uE&t=60s) in order to easily switch between branches, rather than this earlier oversimplification.
